### PR TITLE
fix (stack name duplication for integ tests)

### PR DESCRIPTION
**Pull request by hnishar**
*Issue #, if available:* [183](https://github.com/awslabs/aws-solutions-constructs/issues/183) 
*Description of changes:*  Fixing stack name for integration tests for all constructs

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: b86fcccd504dc0b1a894ce6a1433b1c956e37322
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7677e44a-84bd-4adf-a358-ab834c5a71ce%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/core/lib/utils.ts [hunk 1]
```diff
@@ -123,3 +124,15 @@ export function generateResourceName(
 function removeNonAlphanumeric(s: string) {
   return s.replace(/[^A-Za-z0-9]/g, '');
 }
+
+/**
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/core/lib/utils.ts:128]: This should be in the test folder, not production code (I'm assuming from the name it's only called by tests?)

we already have core/test/test-helper.ts

File: source/patterns/@aws-solutions-constructs/core/test/utils.test.ts [hunk 2]
```diff
@@ -50,3 +50,12 @@ test('Test with non Aphanumeric', () => {
...
   expect(result).toContain('partoneparttwo');
 });
+
+// --------------------------------------------------------------
+// Test generateIntegStackName
+// --------------------------------------------------------------
+test('Test generateIntegStackName', () => {
+  const result = defaults.generateIntegStackName('integ.apigateway-dynamodb-CRUD.js');
+
+  expect(result).toContain('apigateway-dynamodb-CRUD');
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/core/test/utils.test.ts:60]: This is testing test code, which is OK I guess, but not needed.

**Review by biffgaut — CHANGES_REQUESTED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 2b45849ac8fda06adb6e52b19807d892e647631a
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=bf9a9b03-29f4-453a-a292-b252e45e468e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: c027b24c638b6211462194f10e46ae630025a010
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=4f3f073d-508c-4b5f-aeeb-19b38dca83c0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — APPROVED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: b80d5fe4c22d83d96d76c8c253ec2bbf6e5a793b
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=fa1044a0-92d4-456d-a791-2dc187fe5806%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: f0d1abcc91fb01fc268ed5d79d90889128125d95
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=aafd34d3-bebc-473c-8c04-c1620b247d3e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: b5c20ccfb426a84924560a02449b00a6511f2629
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b68884ab-f13c-4ec6-a739-f20c281ef2a0%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 420cc3f5605dde8f0a428c920799f901d1eff643
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=24137ff1-44bf-4ebb-a6f1-0b948c3695ee%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 8c771beab0f30c99d2730664b969c61727b738ec
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1a95356b-f58a-4403-b71b-11097e791c65%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.existingContainer.ts [hunk 3]
```diff
@@ -15,13 +15,14 @@
...
+import { generateIntegStackName } from '@aws-solutions-constructs/core';
 
 // Setup
 const app = new App();
-const stack = new Stack(app, 'test-cloudfront-mediastore');
+const stack = new Stack(app, generateIntegStackName(__filename));
 stack.templateOptions.description = 'Integration test for aws-cloudfront-mediastore with existing mediastore container';
 const mediaStoreContainerObject = new mediastore.CfnContainer(stack, 'MyMediaStoreContainer', {
-  containerName: 'MyMediaStoreContainer'
+  containerName: 'MyMediaStoreContainer2'
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.existingContainer.ts:25]: Why? To avoid a name collision? If so, let's change one or the other name to clarify the difference. Just adding "2" is seldom the best way forward. (It's right up there with Microsoft adding "Ex" to improved API calls).

File: source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.overrideProperties.ts [hunk 4]
```diff
@@ -30,7 +31,10 @@ const mediaStoreContainerProps: mediastore.CfnContainerProps = {
       Effect: 'Allow',
       Principal: '*',
       Action: 'mediastore:*',
-      Resource: '*'
+      Resource: `arn:aws:mediastore:${Aws.REGION}:${Aws.ACCOUNT_ID}:container/MyMediaStoreContainer/*`,
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.overrideProperties.ts:34]: While I agree that this is probably an improvement, we can't just include other functionality in a PR that we happen to be in anyway. Especially a PR with so many changes that this unrelated change will be hard to find. Let's remove this and do it in a separate PR.

File: source/patterns/@aws-solutions-constructs/aws-kinesisstreams-kinesisfirehose-s3/test/integ.existingStreamObj.ts [hunk 5]
```diff
@@ -16,10 +16,11 @@ import { App, Stack, RemovalPolicy } from "@aws-cdk/core";
 import { KinesisStreamsToKinesisFirehoseToS3 } from '../lib';
 import { KinesisStreamsToLambda } from '@aws-solutions-constructs/aws-kinesisstreams-lambda';
 import * as lambda from '@aws-cdk/aws-lambda';
+import { generateIntegStackName } from '@aws-solutions-constructs/core';
 
 // Setup
 const app = new App();
-const stack = new Stack(app, 'test-existing-stream-firehose-s3-stack');
+const stack = new Stack(app, generateIntegStackName(__filename));
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-kinesisstreams-kinesisfirehose-s3/test/integ.existingStreamObj.ts:23]: Wow, this PR must have been drudgery...

File: source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/integ.domain-arguments.ts [hunk 6]
```diff
@@ -27,7 +28,7 @@ const lambdaProps: lambda.FunctionProps = {
 };
 
 const esDomain = 'domain-' + Aws.ACCOUNT_ID;
-const cognitoDomain = 'domainargumentsdomain';
+const cognitoDomain = 'domainargumentsdomain1';
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/integ.domain-arguments.ts:31]: Why? And why "1"? If it must change, is there a better name that clarifies the difference?

**Review by biffgaut — CHANGES_REQUESTED** [review]

File: source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/integ.domain-arguments.ts (hunk 6 above)
Comment (hnishar) [reply source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/integ.domain-arguments.ts:31]: It was failing to deploy in my local envrionment even though the stack was deleted, Cognito service was complaining about the existing domain, I assume some issue with the backend cleanup in Cognito service, I will revisit this change one more time to see if it's giving the same error still

File: source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.overrideProperties.ts (hunk 4 above)
Comment (hnishar) [reply source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.overrideProperties.ts:34]: Writing up a separate PR which will need to merged before this PR!

File: source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.existingContainer.ts (hunk 3 above)
Comment (hnishar) [reply source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.existingContainer.ts:25]: Yep to avoid the name collision

File: source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.overrideProperties.ts (hunk 4 above)
Comment (hnishar) [reply source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.overrideProperties.ts:34]: https://github.com/awslabs/aws-solutions-constructs/pull/203

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: b4a953669330a49516da235b7d4a2213debb9d00
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a282e1d3-7b77-4b94-a9da-7483ce8e9077%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.existingContainer.ts (hunk 3 above)
Comment (hnishar) [reply source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.existingContainer.ts]: Renamed to `MyExistingMediaStoreContainer`

File: source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.overrideProperties.ts (hunk 4 above)
Comment (hnishar) [reply source/patterns/@aws-solutions-constructs/aws-cloudfront-mediastore/test/integ.overrideProperties.ts]: Changed migrated to separate PR and merged into `main`

File: source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/integ.domain-arguments.ts (hunk 6 above)
Comment (hnishar) [reply source/patterns/@aws-solutions-constructs/aws-lambda-elasticsearch-kibana/test/integ.domain-arguments.ts]: Renamed domain name to `mydomainfortesting`

**Review by biffgaut — APPROVED** [review]
